### PR TITLE
Correct cppcheck errors

### PR DESCRIPTION
**Pull request by dinatale2**
### Description

ZFS buildbot STYLE builder was moved to Ubuntu 17.04
which has a newer version of cppcheck. Handle the
new cppcheck errors.

### Motivation and Context

Fix STYLE builder.

### How Has This Been Tested?

Builds locally on a VM.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux code style requirements.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commit messages are properly formatted and contain `Signed-off-by`.
- [ ] Change has been approved by a ZFS on Linux member.

### CODE REVIEW

File: module/zfs/vdev_raidz_math_avx512f.c [hunk 1]
```diff
@@ -241,6 +241,7 @@ typedef struct v {
 }
 
 #define	MUL_x2_DEFINE(x)						\
+/* cppcheck-suppress syntaxError */					\
```
Comment (behlendorf) [inline module/zfs/vdev_raidz_math_avx512f.c]: I think this may need to be,

```
// cppcheck-suppress syntaxError
```

Comment (dinatale2) [reply module/zfs/vdev_raidz_math_avx512f.c]: You're right. I was trying to figure out a way to get the supression comment to be included in the macro so `cppcheck` would actually suppress it and not break builds.

@behlendorf and I discussed offline and I'll have another attempted fix shortly.

**Comment by dinatale2** [comment]
For some reason, having `preprocessorErrorDirective:filename:lineno` won't work... but if I simply ignore all `preprocessorErrorDirective` in `supressions.txt`, `cppcheck` functions as expected.

**Comment by behlendorf** [comment]
That's unfortunate, then how about we exclude these two configs `-UHAVE_SSE2 -UHAVE_AVX512F` instead.

**Comment by dinatale2** [comment]
@behlendorf I went ahead and pinged them on their IRC channel and am waiting for a response. I submitted an updated commit to use `-UHAVE_SSE2 -UHAVE_AVX512F` in the meantime. If I get a response on IRC, I'll update it again.

**Pull request merged**